### ISSUE TITLE
fails to build with `clangd/index/remote/Client.cpp: expected unqualified-id` 

### ISSUE THREAD

**Issue by sylvestre** [llvm/llvm-project#62733]
On Linux, with main:
```console
/build/source/build-llvm/./bin/clang++ -DCLANG_REPOSITORY_STRING="\"++20230516071722+4054c68644df-1~exp1~20230516071738.666\"" -DGOOGLE_PROTOBUF_NO_RTTI=1 -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -Itools/clang/tools/extra/clangd/index/remote -I/build/source/clang-tools-extra/clangd/index/remote -I/build/source/clang-tools-extra/clangd/../include-cleaner/include -Itools/clang/tools/extra/clangd/../clang-tidy -I/build/source/clang/include -Itools/clang/include -Iinclude -I/build/source/llvm/include -I/build/source/clang-tools-extra/clangd -Itools/clang/tools/extra/clangd -fstack-protector-strong -Wformat -Werror=format-security -Wno-unused-command-line-argument -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -ffile-prefix-map=/build/source/build-llvm/tools/clang/stage2-bins=build-llvm/tools/clang/stage2-bins -ffile-prefix-map=/build/source/= -no-canonical-prefixes -fno-common -Woverloaded-virtual -Wno-nested-anon-types -O2 -DNDEBUG -g1  -fno-exceptions -funwind-tables -std=c++17 -MD -MT tools/clang/tools/extra/clangd/index/remote/CMakeFiles/obj.clangdRemoteIndex.dir/Client.cpp.o -MF tools/clang/tools/extra/clangd/index/remote/CMakeFiles/obj.clangdRemoteIndex.dir/Client.cpp.o.d -o tools/clang/tools/extra/clangd/index/remote/CMakeFiles/obj.clangdRemoteIndex.dir/Client.cpp.o -c /build/source/clang-tools-extra/clangd/index/remote/Client.cpp
/build/source/clang-tools-extra/clangd/index/remote/Client.cpp:71:48: error: expected unqualified-id
    trace::Span Tracer(RequestT::descriptor()->name());
                                               ^
/build/source/clang-tools-extra/clangd/index/remote/Client.cpp:71:17: error: no matching constructor for initialization of 'trace::Span'
    trace::Span Tracer(RequestT::descriptor()->name());
                ^
/build/source/clang-tools-extra/clangd/index/remote/Client.cpp:133:5: note: in instantiation of function template specialization 'clang::clangd::remote::(anonymous namespace)::IndexClient::streamRPC<clang::clangd::remote::LookupRequest, clang::clangd::remote::LookupReply, clang::clangd::LookupRequest, llvm::function_ref<void (const clang::clangd::Symbol &)>>' requested here
    streamRPC(Request, &remote::v1::SymbolIndex::Stub::Lookup, Callback);
    ^
```

**Comment by kadircet** [comment]
Hi @sylvestre could you mention the version for `/build/source/build-llvm/./bin/clang++`, revision of llvm-checkout and protobuf library versions?

**Comment by sylvestre** [comment]
it is a stage 2 build, so, same revision as clangd 
So, this changeset: 4054c68644df
protobuf 3.12.4-1

Full log:
https://llvm-jenkins.debian.net/job/llvm-toolchain-bullseye-binaries/architecture=arm64,distribution=bullseye,label=arm64/560/consoleFull

**Comment by kadircet** [comment]
should be fixed with d4a6e4c1eec0f2634febd15725d921d4a25d47e7
